modules/c++/sys/source/OSUnix.cpp

```diff
@@ -189,9 +188,14 @@ std::string sys::OSUnix::getTempName(const std::string& path,
     else
     {
         name = tempname;
+        sys::File (name, sys::File::READ_AND_WRITE, sys::File::CREATE);
```

Comment:
One more pedantic thing.  Things could throw here if either 1. The assignment into `name` fails (`std::string` ran out of memory) or 2. The file creation in `sys::File` fails.  In these scenarios, we will not get a chance to free `tempname` and it will leak.  Can you make an RAII class in the unnamed namespace that takes ownership of a `char*` and frees it in its destructor to guard against this?

Also technically the `sys::File` only needs to be opened with write permissions - don't need to read it